sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/ExpressionImplUtils.java

```diff
@@ -35,6 +35,35 @@ public class ExpressionImplUtils {
   private static final int GCM_IV_LEN = 12;
   private static final int GCM_TAG_LEN = 128;
 
+  /**
+   * Function to check if a given number string is a valid Luhn number
+   * @param numberString
+   *  the number string to check
+   * @return
+   *  true if the number string is a valid Luhn number, false otherwise.
+   */
+  public static boolean isLuhnNumber(UTF8String numberString) {
+    String digits = numberString.toString();
+    // Check if all characters in the input string are digits.
+    if (!digits.isEmpty() && digits.chars().allMatch(Character::isDigit)) {
```

Comment:
As @srowen mentioned, how about returning false in the for loop if space is found? the code can be
```
if (digits.isEmpty()) return false;
for ...
  if (!Character.isDigit(digits.charAt(i))) return false;
  ...
```

Reply:
Done